WeakAuras/Types.lua

```diff
@@ -266,12 +266,21 @@ local simpleFormatters = {
   },
 }
 
----@type table<string, {display: string, AddOptions: fun(symbol: string, hidden: boolean, addOption: fun(string, table), get: fun(string): any), CreateFormatter: fun(symbol: string, get: fun(string): any): fun(value: any, state: table): any}>
+---@alias optionAdder fun(name: string, option: table)
+---@alias optionGetter fun(name: string): any
+---@alias formatter fun(input: string): string, nil | boolean
+
+---@class FormatType
+---@field display string
+---@field AddOptions fun(symbol: string, hidden: boolean, addOption: optionAdder, get: optionGetter)
+---@field CreateFormatter fun(symbol: string, get: optionGetter, withoutColor: boolean, data: table): formatter | nil
+
+---@type table<string, FormatType>
 Private.format_types = {
```

Comment:
If you checkout this branch, you might notice 2 diagnostics for redundant-return-value, related to this type annotation. They're on these lines, which I can't make sense of: https://github.com/WeakAuras/WeakAuras2/blob/b314b97d439c5b80f7926af4d9c3da76fb565688/WeakAuras/Types.lua#L492-L494